### PR TITLE
Adjust proxies for changed ChannelID

### PR DESCRIPTION
**Pull request by err508**
As implied by [current](https://github.com/raiden-network/raiden-contracts/pull/198) changes of  signatures of contract functions and some helpers (As described in #1940 and specified [here](https://github.com/raiden-network/raiden-contracts/issues/193))

What has been done w.r.t. #1940: 
- `ChannelID` type from `bytes` to `int`
- `detail_participant` and `detail_channel` have been changed to work with the new channel identifier
(as well as functions composed from those, i.e. `detail_participants`, `detail`)

- Added an overflow/underflow check for the `channel_identifier`
- added appropriate arguments to `set_total_deposit` function
- inverted the argument order of `balance_hash/nonce` in close
to be consistent with the contracts (optional)
- inverted the argument order of `balance_hash/nonce` in update_transfer
to be consistent with the contracts (optional)
- added `channel_identifier` to `unlock`

-  added `channel_identifier` + `partner` to
`token_network.setTotalDeposit` call inside `set_total_deposit`
- reordered `close` arguments to be consistent with contracts (optional)
- reordered `update_transfer` arguments to be consistent with contracts (optional)
- added `channel_identifer` to `token_network.updateNonClosingBalanceProof `call
inside update_transfer
- added `channel_identifier` to `token_network.unlock` call inside `unlock`
- added `channel_identifier` to `token_network.settleChannel` call inside `settle`
for both cases

### CODE REVIEW

**Comment by LefterisJP** [comment]
As discussed with @err508 in rocketchat, I will open today a PR with the initial work I did on #2000 and he will rebase his work on mine to avoid stepping on each other's toes as my PR will touch a large part of the codebase and the tests.

**Comment by LefterisJP** [comment]
Thank you for your work @err508. I am now going to close this in favour of the WIP https://github.com/raiden-network/raiden/pull/2035 PR. I want to continue on it on my own now as we would be touching similar code and it would only lead to conflicts.

**Comment by LefterisJP** [comment]
Note: @err508 Even though I rebased my work on yours, I did not review it and as such I am probably going to end up changing some of the things. Will leave comments here in this PR if something is off.

File: raiden/network/proxies/payment_channel.py [hunk 1]
```diff
@@ -26,6 +27,10 @@ def __init__(
             token_network: TokenNetwork,
             channel_identifier: typing.ChannelID,
     ):
+
+        if 0 > channel_identifier > UINT64_MAX:
```
Comment (LefterisJP) [inline raiden/network/proxies/payment_channel.py]: `channel identifier` is a uint256 not a uint64. Check [here](https://github.com/raiden-network/raiden-contracts/blob/f5e106591c7738f70e14e4975b109fd0825d5816/raiden_contracts/contracts/TokenNetwork.sol#L32).

Also this check does not do what you think it does. Don't chain comparison operators like this without being sure what they are doing. This is equivalent to `if channel_identifier < 0 and channel_identifier > UINT64_MAX:` which is always False, even when you intended it to be true.

File: raiden/network/proxies/payment_channel.py [hunk 2]
```diff
@@ -26,6 +27,10 @@ def __init__(
...
+            raise ValueError
```
Comment (LefterisJP) [inline raiden/network/proxies/payment_channel.py]: If you have a valueError provide a reason for the error in the ctor.

File: raiden/network/proxies/payment_channel.py [hunk 3]
```diff
@@ -121,8 +127,8 @@ def close(
         self.token_network.close(
             channel_identifier=self.channel_identifier,
             partner=self.participant2,
-            nonce=nonce,
             balance_hash=balance_hash,
+            nonce=nonce,
```
Comment (LefterisJP) [inline raiden/network/proxies/payment_channel.py]: @err508 Why did you change the position of the nonce argument here and below?

File: raiden/network/proxies/token_network.py [hunk 4]
```diff
@@ -167,7 +167,7 @@ def new_netting_channel(
             'new_netting_channel called',
             peer1=pex(self.node_address),
             peer2=pex(partner),
-            channel_identifier=encode_hex(channel_identifier),
+            channel_identifier=str(channel_identifier),
```
Comment (LefterisJP) [inline raiden/network/proxies/token_network.py]: this is not needed for the logs. Int is turned into string by the log itself.

File: raiden/network/proxies/token_network.py [hunk 5]
```diff
@@ -530,8 +542,8 @@ def close(
             self,
             channel_identifier: typing.ChannelID,
             partner: typing.Address,
-            nonce: typing.Nonce,
             balance_hash: typing.BalanceHash,
+            nonce: typing.Nonce,
```
Comment (LefterisJP) [inline raiden/network/proxies/token_network.py]: also here why the change of the nonce argument position?

File: raiden/network/proxies/token_network.py [hunk 6]
```diff
@@ -590,8 +602,8 @@ def update_transfer(
...
```
Comment (LefterisJP) [inline raiden/network/proxies/token_network.py]: same here. why the change of the nonce argument position?

File: raiden/network/proxies/token_network.py [hunk 7]
```diff
@@ -114,7 +114,7 @@ def new_netting_channel(
             settle_timeout: The settle timout to use for this channel.
 
         Returns:
-            The address of the new netting channel.
+            The ChannelID (int) of the new netting channel.
```
Comment (LefterisJP) [inline raiden/network/proxies/token_network.py]: stating the type of the channel id in the comment is not needed. It's redundant information. (if god forbid we ever needed to change it again that comment would be wrong)

File: raiden/network/proxies/payment_channel.py (hunk 3 above)
Comment (err508) [reply raiden/network/proxies/payment_channel.py]: I reordered the arguments according to the contracts functions arguments.

Comment (LefterisJP) [reply raiden/network/proxies/payment_channel.py]: I see but this is not needed. The arguments need to be the same only for the contract proxy call, for which they are and to which you did not add the `channel_identifier`.

File: raiden/network/proxies/token_network.py (hunk 5 above)
Comment (LefterisJP) [reply raiden/network/proxies/token_network.py]: This function here had the nonce order changed but you did not add the `channel_identifier` [here](https://github.com/raiden-network/raiden/pull/2018/files#diff-d02574a7a7eca826412780e80a6326a3R581) into the actual proxy call.

File: raiden/network/proxies/payment_channel.py (hunk 3 above)
Comment (LefterisJP) [reply raiden/network/proxies/payment_channel.py]: To elaborate a bit. The reason I say it's not needed is that perhaps these functions are used in other places which you have not seen and things broke for no reason.

In principle, when making any changes to a codebase aim for the minimal working changes as every change, no matter how small, can have unforeseen consequences

Comment (err508) [reply raiden/network/proxies/payment_channel.py]: As stated in the commit message, these changes are optional. I'm aware it might require to change a lot, but I also found it annoying to compare function signatures with signatures that inconsistently ordered inside and across repos.

Comment (LefterisJP) [reply raiden/network/proxies/payment_channel.py]: I understand the sentiment. But this is how software evolves and keeping all functions have the same argument order is just something that's not going to happen. No gains and too much time and effort to maintain.

Comment (err508) [reply raiden/network/proxies/payment_channel.py]: I'd like to "leave the code cleaner than it was", as we have to touch all functions anyhow

Comment (LefterisJP) [reply raiden/network/proxies/payment_channel.py]: > I'd like to "leave the code cleaner than it was"

You did not. You changed a function but you did not change all the places that function is called from. So you left the code broken and I had to [add additional fixes](https://github.com/raiden-network/raiden/pull/2035/commits/596353b06e75aad242f11f426f14e384f8aa2635#diff-295b7685bc1e9c921272d10d5028abb8R349) on top.

And we don't have to touch all functions. There was no need to touch the order of the arguments of the functions of the proxies themselves.

Anyway it's done and probably fixed now. I am not discussing this further. In the future you will keep all code changes minimal.
